src/NServiceBus.Core/TransportDispatch/DispatchTerminator.cs

```diff
@@ -25,7 +25,7 @@ public override void Terminate(DispatchContext context)
             }
             var routingStrategy = context.GetRoutingStrategy();
 
-            dispatchStrategy.Dispatch(dispatcher, context.Get<OutgoingMessage>(), routingStrategy, context.GetConsistencyGuarantee(), context.GetDeliveryConstraints(), context);
+            dispatchStrategy.Dispatch(dispatcher, context.Get<OutgoingMessage>(), routingStrategy, context.GetConsistencyGuarantee(), context.GetDeliveryConstraints(), context).GetAwaiter().GetResult();
```

Comment:
General question - whenever we dispatch, could that could be safely assumed as an IO operation?